samples/nrf9160/nrf_cloud_mqtt_multi_service/src/location_tracking.c

```diff
@@ -47,26 +48,56 @@ static void gnss_antenna_configure(int ret, void *ctx)
 }
 NRF_MODEM_LIB_ON_INIT(location_tracking_init_hook, gnss_antenna_configure, NULL);
 
-void location_agps_data_handler(const char *buf, size_t len)
+void location_assistance_data_handler(const char *buf, size_t len)
 {
-	/* We don't actually check whether the passed-in payload contains AGPS data!
-	 * Instead, nrf_cloud_agps_process() will check for us, ignoring payloads that
-	 * don't contain AGPS data, and processing (and sending to the modem) payloads that do.
+	/* We don't actually check whether the passed-in payload contains assistance data!
+	 * Instead, nrf_cloud_agps_process and nrf_cloud_pgps_process will check for us,
+	 * ignoring payloads that don't contain assistance data, and processing
+	 * (and sending to the modem) payloads that do.
+	 *
+	 * Note that P-GPS payloads are actually just resource identifiers encoded with JSON,
+	 * and upon encountering this resource identifier, nrf_cloud_pgps_process will
+	 * automatically download and store the predictive data it references.
+	 *
+	 * A-GPS payloads, on the other hand, are pure binary, and are passed more or less directly
+	 * to the modem by nrf_cloud_agps_process.
 	 */
+	int err;
 
-	int err = nrf_cloud_agps_process(buf, len);
+	/* First, try to process the payload as A-GPS data, if AGPS is enabled */
+	if (IS_ENABLED(CONFIG_NRF_CLOUD_AGPS)) {
+		err = nrf_cloud_agps_process(buf, len);
+		if (err) {
+			if (err != -EBADMSG) {
+				LOG_WRN("Unable to process A-GPS data, error: %d", err);
+			} else {
+				/* The passed in MQTT message was likely just not AGPS-related. */
+				LOG_DBG("Failed to extract A-GPS data from passed in MQTT message, "
+					"got -EBADMSG. This is normal behavior if a non-AGPS "
+					"message was received.");
+			}
+		} else {
+			LOG_DBG("A-GPS data processed");
+			return;
+		}
+	}
 
-	if (err) {
-		if (err != -EBADMSG) {
-			LOG_WRN("Unable to process A-GPS data, error: %d", err);
+	/* Failing that, try to process the payload as P-GPS data, if P-GPS is enabled */
+	if (IS_ENABLED(CONFIG_NRF_CLOUD_PGPS)) {
+		err = nrf_cloud_pgps_process(buf, len);
+		if (err) {
+			if (err != -EBADMSG && err != -EFTYPE) {
+				LOG_WRN("Unable to process P-GPS data, error: %d", err);
+			} else {
+				/* The passed in MQTT message was likely just not AGPS-related. */
```

Comment:
PGPS-related

Reply:
oh, and then Justin made the same comment but as a change request

Reply:
No idea how I managed to not include my correction for this in my last push, thank you